populate_example_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fx_test
+
+import (
+	"fmt"
+
+	"go.uber.org/fx"
+)
+
+func ExamplePopulate() {
+	// Some external module that provides a user name.
+	type Username string
+	UserModule := fx.Provide(func() Username { return "john" })
+
+	// We want to get the user out of the dependency injection container.
+	var user Username
+	fx.New(
+		UserModule,
+
+		fx.Populate(&user),
```

Comment:
If we're showing a working example, we should have Start/Stop in there, but
doing that will make the example less readable IMO.

It may be preferable to have usage in the comments for Populate instead
because it lets us make assumptions about the rest of the code.

```
//   var conn *sql.DB
//   app := fx.New(
//     database.Module,
//     fx.Populate(&conn),
//   )
//   if err := app.Start(ctx); err !
//     return err
//   }
```

Thoughts?

Reply:
Hmm I like the idea of an example showing up in the godoc nicely, I can add start/stop at the end with comments.